tests/unit/distributed/remote_arrays_addition.cpp

```diff
@@ -0,0 +1,191 @@
+    {
+        std::int64_t locality_ =
+            hpx::naming::get_locality_id_from_id(hpx::find_here());
+        return hpx::make_ready_future(primitive_argument_type(locality_));
+    }
+}}}
+
+///////////////////////////////////////////////////////////////////////////////
+bool is_locality_0 = false;
+
+std::string init_arrays = R"(block(
+    define(arrays_init,
+        block(
+            define(a, constant(41.0, list(2, 4))),
+            define(rowsA, shape(a, 0)),
+            define(columnsA, shape(a, 1)),
+            define(a1, slice(a, list(0, rowsA), list(0, columnsA / 2))),
+            define(a2, slice(a, list(0, rowsA), list(columnsA / 2, columnsA))),
+            // cout(a),
+            // cout(a1),
+            // cout(a2),
+
+            define(b, constant(1.0, list(2, 4))),
+            define(rowsB, shape(b, 0)),
+            define(columnsB, shape(b, 1)),
+            define(b1, slice(b, list(0, rowsB), list(0, columnsB / 2))),
+            define(b2, slice(b, list(0, rowsB), list(columnsB / 2, columnsB))),
+
+            // cout(b),
+            // cout(b1),
+            // cout(b2),
+            list(a1, a2, b1, b2)
+        )
+    ),
+    arrays_init
+))";
+
+std::string code1 = R"(block(
+    define(code1, arg0, arg1,
+        block(
+            define(c1, __add(arg0, arg1)),
+            debug(c1)
+        )
+    ),
+    code1
+))";
+
+std::string code2 = R"(block(
+    define(code2, arg0, arg1,
+        block(
+            define(c2, __add(arg0, arg1)),
+            debug(c2)
+        )
+    ),
+    code2
+))";
+
+///////////////////////////////////////////////////////////////////////////////
+phylanx::execution_tree::compiler::function compile(
+    std::string const& codestr, std::uint32_t locality_id)
+{
+    phylanx::execution_tree::compiler::function_list snippets;
+    phylanx::execution_tree::compiler::environment env =
+        phylanx::execution_tree::compiler::default_environment(
+            hpx::naming::get_id_from_locality_id(locality_id));
+
+    auto const& code = phylanx::execution_tree::compile(codestr, snippets, env);
+    return code.run();
+}
+
+phylanx::execution_tree::primitive_argument_type compile_and_run(
+    std::string const& codestr)
+{
+    phylanx::execution_tree::compiler::function_list snippets;
+    phylanx::execution_tree::compiler::environment env =
+        phylanx::execution_tree::compiler::default_environment();
+
+    auto const& code = phylanx::execution_tree::compile(codestr, snippets, env);
+    return code.run();
+}
+
+void test_remote_arrays_chain(std::uint32_t here, std::uint32_t there)
+{
+    auto temp = compile_and_run(init_arrays);
+    auto arrays_list = temp();
+    auto arrays_list_value =
+        phylanx::execution_tree::extract_list_value(arrays_list);
+    HPX_TEST_EQ(arrays_list_value.size(), 4);
+    auto it = arrays_list_value.begin();
+    auto a1 = *it++;
+    auto a2 = *it++;
+    auto b1 = *it++;
+    auto b2 = *it++;
+
```

Comment:
`arrays_list_value` is a `std::vector<primitive_argument_type>`. Why not simply use the indexing operator to access its elements?